### PR TITLE
Make SSL_clear_options pass new options to record layer

### PR DESCRIPTION
**Pull request by lan1120**
Make SSL_clear_options pass new options to record layer, so we can more flexibly control certain features, such as DTLS record replay check.

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

File: ssl/ssl_lib.c [hunk 1]
```diff
@@ -6044,7 +6045,16 @@ uint64_t SSL_clear_options(SSL *s, uint64_t op)
...
 
-    return sc->options &= ~op;
+    sc->options &= ~op;
+
+    *opts++ = OSSL_PARAM_construct_uint64(OSSL_LIBSSL_RECORD_LAYER_PARAM_OPTIONS,
+                                          &sc->options);
+    *opts = OSSL_PARAM_construct_end();
+
+    /* Ignore return value */
+    sc->rlayer.rrlmethod->set_options(sc->rlayer.rrl, options);
```
Comment (mattcaswell) [inline ssl/ssl_lib.c:6055]: Shouldn't we be calling this on the wrlmethod too?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: ssl/ssl_lib.c (hunk 1 above)
Comment (lan1120) [reply ssl/ssl_lib.c:6056]: Done. I also added it to SSL_set_options function.

File: ssl/ssl_lib.c [hunk 2]
```diff
@@ -6044,7 +6046,17 @@ uint64_t SSL_clear_options(SSL *s, uint64_t op)
...
+    sc->rlayer.rrlmethod->set_options(sc->rlayer.wrl, options);
```
[the remark this answers is not in the document]
Comment (lan1120) [reply ssl/ssl_lib.c]: Done.

**Review by hlandau — APPROVED** [review]

**Comment by hlandau** [comment]
Needs reapproval

**Review by mattcaswell — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed to master and 3.2. Thanks.

**Pull request closed without merging**